pkg/sass-parser/lib/src/expression/color.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2025 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as postcss from 'postcss';
+import * as sass from 'sass';
+
+import {LazySource} from '../lazy-source';
+import {NodeProps} from '../node';
+import {RawWithValue} from '../raw-with-value';
+import type * as sassInternal from '../sass-internal';
+import * as utils from '../utils';
+import {Expression} from '.';
+
+/**
+ * The initializer properties for {@link ColorExpression}.
+ *
+ * @category Expression
+ */
+export interface ColorExpressionProps extends NodeProps {
+  raws?: ColorExpressionRaws;
+
+  /**
+   * The color value for the expression. This must be in the RGB color space,
+   * or {@link ColorExpression} will throw an error.
+   */
+  value: sass.SassColor;
+}
+
+/**
+ * Raws indicating how to precisely serialize a {@link ColorExpression}.
+ *
+ * @category Expression
+ */
+export interface ColorExpressionRaws {
+  /**
+   * The raw string representation of the color.
+   *
+   * Colors can be represented as named keywords or as hex codes, with or
+   * without capitalizations and with a varying number of digits.
+   */
+  value?: RawWithValue<sass.SassColor>;
+}
+
+/**
+ * An expression representing a color literal in Sass.
+ *
+ * @category Expression
+ */
+export class ColorExpression extends Expression {
+  readonly sassType = 'color' as const;
+  declare raws: ColorExpressionRaws;
+
+  /**
+   * The color represented by this expression. This will always be a color in
```

Comment:
Why will it always be an RGB color? It doesn't seem like it's a limitation on sass_api 🤔

Is this a temporary limitation?

Reply:
Because Sass has no literal syntactic representation of non-RGB colors. Every other color type is returned by function calls.